program_transformers/src/token_extensions/mod.rs

```diff
@@ -195,7 +196,11 @@ pub async fn handle_token_extensions_program_account<'a, 'b, 'c>(
             txn.commit().await?;
 
             if let Some(metadata) = &extensions.metadata {
-                upsert_asset_data(metadata, account_key.clone(), slot, db).await?;
+                if let Ok(info) = upsert_asset_data(metadata, account_key.clone(), slot, db).await {
+                    download_metadata_notifier(info)
+                        .await
+                        .map_err(ProgramTransformerError::DownloadMetadataNotify)?;
+                }
```

Comment:
```suggestion
                upsert_asset_data(metadata, account_key.clone(), slot, db)
                  .then(download_metadata_notifier)
                  .await
                  .map_err(ProgramTransformerError::DownloadMetadataNotify)?;
                }
```
I think you can chain those calls.

Reply:
as we have an Option now it can be dificult to read wyt ?